### DISCUSSION TITLE
How do i implement the "Add to Cart" functionality?

### DISCUSSION THREAD

**Discussion by dianaow** [pevey/sveltekit-medusa-starter#10] [Q&A]
I have set up the starter and Medusa backend, but the api endpoints '/store/carts' or '/store/carts/${locals.cartid}/line-items/' do not exist. Do i have to extend the Medusa backend by customizing these carts routes and serviece?

**Comment by pevey** [comment]
The Medusa backend may have changed since this project was last updated.  I no longer use Medusa, so I don't currently have plans to keep this starter up to date with the frequent changes made to the Medusa backend.  I now use and recommend Vendure.

**Comment by dianaow** [comment]
Thanks for the response! I thought to just have a try at Medusa first. Also, i'm not familiar with GraphQL at all, so I thought Vendure wouldn't suit.  

I mistook the reason for the error initially. The response for '/store/carts' was 400, which means that the endpoint exist but there was an error in sending the POST request. I could make the POST request successfully with Postman, but from the storefront upon click of the 'add to cart' button, an error happens inside sveltekit-superfetch.

<img width="1033" alt="Screenshot 2024-02-29 at 3 03 43 PM" src="https://github.com/pevey/sveltekit-medusa-starter/assets/33983719/c69fdde2-cf03-40eb-9166-40d7714eca07">

**Comment by dianaow** [comment]
Resolved by connecting to the medusa client direclty instead of using the "sveltekit-medusa-client" library. Your code was still useful to understand what APIs to connect to and how, thanks!
